react/internal/http-switch-link.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * This file defines a linker middleware used for changing http verbs, allowing
+ * cache in CDNs and other http-based caching content delivery networks
+ *
+ * If the query type is Query, the request will be made using an HTTP GET
+ * If the query type is Mutation, the request will be made using HTTP POST
+ *
+ */
+import {ApolloLink} from 'apollo-link'
+import {parse, format} from 'url'
+import {print} from 'graphql'
+
+const isEmpty = (val) => (val === undefined || val === "" || val === {} || val === [])
```

Comment:
`val === {}` will always evaluate to false because `{}` declares a new object and `===` checks for referential equality.

Reply:
Same for array.

Reply:
(Take a look at the reference values section here: http://firstdoit.github.io/js-closures-workshop/)